src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -1,32 +1,76 @@
+import com.lowagie.text.{Document, Element, HeaderFooter, Rectangle}
 import com.lowagie.text.pdf.PdfWriter
 import edg_ide.swing.ElkNodePainter
+import edgir.elem.elem.{BlockLike, HierarchyBlock}
 import org.eclipse.elk.graph.ElkNode
+import edg.util.SeqMapSortableFrom._
+import edg.wir.{DesignPath, ProtoUtil}
+import edg_ide.edgir_graph.HierarchyGraphElk
 
 import java.awt.Color
-import java.io.FileOutputStream
+import java.io.{FileNotFoundException, FileOutputStream, IOException}
 
 
 object PDFGeneratorUtil{
 
-  def generate(rootNode: ElkNode, fileName: String): Unit = {
-    // TODO: Set a fixed document size and scale graphics accordingly?
-    val width = rootNode.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val height = rootNode.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val document = new Document(new Rectangle(width, height))
+  private def generatePageSize(node: ElkNode): (Float, Float) = {
+    val width = node.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
+    val height = node.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
+    (width, height)
+  }
+
+  def generate(content: HierarchyBlock, fileName: String): Unit = {
+    try {
+      val document = new Document()
+      val writer = PdfWriter.getInstance(document, new FileOutputStream(fileName))
+      val header = new HeaderFooter(true)
+      header.setBorder(Rectangle.NO_BORDER)
+      header.setAlignment(Element.ALIGN_RIGHT)
+      document.setFooter(header)
+
+      def printNode(node: ElkNode): Unit = {
+        val (width, height) = generatePageSize(node)
+        document.setPageSize(new Rectangle(width, height))
+
+        if (document.isOpen) {
+          document.newPage
+        }
+        else {
+          document.open()
+        }
+
+        val cb = writer.getDirectContent
+        val graphics = cb.createGraphics(width, height)
+        val painter = new ElkNodePainter(node)
+        painter.paintComponent(graphics, Color.white)
+        graphics.dispose()
+      }
+
+      def printNextHierarchyLevel(block: HierarchyBlock, path: DesignPath = DesignPath()): Unit = {
+        val node = HierarchyGraphElk.HBlockToElkNode(block, path)
+        printNode(node)
 
-    // TODO: make a try..catch for FileOutputStream
-    val writer = PdfWriter.getInstance(document, new FileOutputStream(fileName))
-    document.open()
-    val cb = writer.getDirectContent
-    val graphics = cb.createGraphics(width, height)
+        val nameOrder = ProtoUtil.getNameOrder(block.meta)
+        val children: Map[DesignPath, HierarchyBlock] = block.blocks.map {
+          case (name, subblock) => (name, subblock.`type`)
+        }.sortKeysFrom(nameOrder)
+          .collect {
+            case (name, BlockLike.Type.Hierarchy(subblock)) if subblock.blocks.nonEmpty => (path + name, subblock)
+          }.toMap
 
-    val painter = new ElkNodePainter(rootNode)
-    painter.paintComponent(graphics, Color.white)
+        children.foreach(hBlock => {
```

Comment:
```suggestion
        children.foreach { case (path, subblock) => {
```

Using a case statement to unpack the element and give them meaningful names can be a more readable than `._1` and `._2` accessors.